src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -990,18 +1214,416 @@ partial class C
+            var source = """
+                partial class C
+                {
+                    private partial int P1 { get; set; }
+                    partial int P2 { get; set; }
+                }
+
+                partial class C
+                {
+                    partial int P1 { get => throw null!; set { } }
+                    private partial int P2 { get => throw null!; set { } }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (9,17): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+                //     partial int P1 { get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "P1").WithLocation(9, 17),
+                // (10,25): error CS8799: Both partial member declarations must have identical accessibility modifiers.
+                //     private partial int P2 { get => throw null!; set { } }
+                Diagnostic(ErrorCode.ERR_PartialMemberAccessibilityDifference, "P2").WithLocation(10, 25));
+        }
+
+        [Fact]
+        public void TypeDifference_01()
+        {
+            var source = """
+                partial class C
+                {
+                    partial int P { get; set; }
+                    partial string P { get => ""; set { } }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (4,20): error CS9307: Both partial property declarations must have the same type.
+                //     partial string P { get => ""; set { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyTypeDifference, "P").WithLocation(4, 20));
+        }
+
+        [Fact]
+        public void TypeDifference_02()
+        {
+            var source = """
+                #nullable enable
+                partial class C
+                {
+                    partial string? P1 { get; set; }
+                    partial string P1 { get => ""; set { } }
+
+                    partial string P2 { get; set; }
+                    partial string? P2 { get => ""; set { } }
+
+                    partial string?[] P3 { get; set; }
+                    partial string[] P3 { get => []; set { } }
+
+                    partial string[] P4 { get; set; }
+                    partial string?[] P4 { get => []; set { } }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (5,20): error CS9308: Partial property declarations 'string? C.P1' and 'string C.P1' have signature differences.
+                //     partial string P1 { get => ""; set { } }
+                Diagnostic(ErrorCode.WRN_PartialPropertySignatureDifference, "P1").WithArguments("string? C.P1", "string C.P1").WithLocation(5, 20),
```

Comment:
>WRN_PartialPropertySignatureDifference [](http://example.com/codeflow?start=37&length=38)

We probably need to handle nullability differences specially. Notice that `WRN_NullabilityMismatchInReturnTypeOnPartial` is part of `ErrorFacts.NullableWarnings`, which means that the warnings can be suppressed with `/nowarn:nullable`. I think we want the same behavior here

Reply:
Partial methods also have an issue where safe nullability differences are not reported using a diagnostic in the NullableWarnings group.

```cs
#nullable disable warnings

partial class C {
    public partial string? M();
    public partial string M2();
}

partial class C {
    public partial string M() => ""; // warning CS8826: Partial method declarations 'string? C.M()' and 'string C.M()' have signature differences
    public partial string? M2() => ""; // no warning
}
```

I'm unsure whether to do anything about that.

Reply:
Yup, that partial method behavior does seem wrong too.

I think the solution is to relax the PartialMethodStrictComparer to ignore nullability differences.
Partial methods already check for unsafe nullability differences (with `CheckValidNullableMethodOverride`) and report the dedicated warning `WRN_NullabilityMismatchInReturnTypeOnPartial` which is part of the nullable warnings set.
The same approach would work for partial properties (check for nullability differences separately, then do a stricter comparison ignoring nullability to catch other unchecked differences).

Reply:
> I think the solution is to relax the PartialMethodStrictComparer to ignore nullability differences.

We would also need a new diagnostic, probably in the same wave as `WRN_PartialMethodSignatureDifference`, in order to continue reporting safe nullability differences.

Reply:
Why report safe nullability differences at all?

Reply:
LDM decided that there is value in reporting a diagnostic for any signature difference between the parts of a partial method, except for oblivious nullable differences. My goal is to carry that decision through to this feature and not to attempt to change that decision.

https://github.com/dotnet/csharplang/blob/main/meetings/2020/LDM-2020-09-14.md#partial-method-signature-matching

Reply:
Thanks for digging up the LDM notes :heart: 
Okay to keep as-is then (unsuppressable nullability diagnostic).

Reply:
nit: Consider including a test that shows the warning cannot be suppressed